### PR TITLE
#1066 Add any numbers of documents to exercise downloads

### PR DESCRIPTION
**Pull request by lloback**
- created the popup for uploading files
- added the possibility of adding / removing/ replacing
- commmented out the tests
- validation

### CODE REVIEW

**Review by mbrookeswebdev — APPROVED** [review]

**Review by warrensearle — CHANGES_REQUESTED** [review]
This is looking good Mr Loback 👍 
Only changes are UI/UX tweaks we discussed such as moving the links, including some instructions and probably removing the Upload Files sub-title.
e.g. (only with better column widths)
Screenshot 2020-12-14 at 10 30 13
Ooh, and perhaps have a 'Required' or 'Optional' lozenge next to each of the file types which gets replaced with 'Done'

File: src/views/Exercises/Edit/Downloads.vue [hunk 1]
```diff
@@ -145,6 +231,65 @@ export default {
...
+          done: true,
+        },
+        { 
+          title: 'Welsh Translation Upload', 
+          id: 'welshTranslation',
+          name: 'welsh-translation',
+          done: true,
+        },
+        { 
+          title: 'Other Donwloads Upload',
```
Comment (tomlovesgithub) [inline src/views/Exercises/Edit/Downloads.vue:285]: little typo here - donwload

File: src/views/Exercises/Edit/Downloads.vue [hunk 2]
```diff
@@ -1,14 +1,93 @@
...
+                    >
+                      <a
+                        class="govuk-link moj-download-link"
+                        href="#"
+                        @click.prevent="modalUploadOpen({ ...upload, fileRef: item.file, fileTitle: item.title })"
+                      >
+                        {{ item.title }}
+                      </a>
+                    </span>
+                    <span :id="`${upload.id}-error`" :ref="`${upload.id}-error`" class="govuk-error-message"><span class="govuk-visually-hidden">Error:</span> Please upload {{ upload.id }} </span>
```
Comment (tomlovesgithub) [inline src/views/Exercises/Edit/Downloads.vue:51]: my linting wanted this split up into multiple lines

**Review by tomlovesgithub — CHANGES_REQUESTED** [review]
Functionally working fine for me, few small errors that i can see, other than that looks to meet criteria nicely

File: src/views/Exercises/Edit/Downloads.vue (hunk 2 above)
Comment (lloback) [reply src/views/Exercises/Edit/Downloads.vue]: fixed

Comment (lloback) [reply src/views/Exercises/Edit/Downloads.vue]: linted.

File: src/views/Exercises/Edit/Downloads.vue (hunk 1 above)
Comment (lloback) [reply src/views/Exercises/Edit/Downloads.vue]: corrected

**Review by warrensearle — APPROVED** [review]
Looks good to me 👍

**Pull request merged**